pkg/util/port/portlist/poller_linux_macos.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2014-present Datadog, Inc.
+
+//go:build darwin || linux
+
+package portlist
+
+import "sort"
+
+func (a *Port) lessThan(b *Port) bool {
```

Comment:
These two functions are agnostic to operating system, is that right? If so I would move them to a source file that doesn't use any build tags.

Reply:
I originally had them in `poller.go` but then the code would fail to compile on Windows because the functions aren't used. Is the best practice for this kind of scenario to call the functions anyway in a Windows specific implementation?

Reply:
I think this problem comes from the linter that complains these functions are not used on windows (since there's no windows implementation yet). This can be solved by adding unit tests.

Reply:
Unit tests added in `poller_test.go`